contracts/pm/TicketBroker.sol

```diff
@@ -129,6 +133,17 @@ contract TicketBroker {
         emit SignersApproved(msg.sender, _signers);
     }
 
+    function requestSignersRevocation(address[] _signers) public {
+        Sender storage sender = senders[msg.sender];
+        uint256 revocationBlock = block.number + signerRevocationPeriod;
+
+        for (uint256 i = 0; i < _signers.length; i++) {
+            sender.signersRevocationBlock[_signers[i]] = revocationBlock;
```

Comment:
Should we set the revocation block for a signer address that was never approved? I lean towards no and skipping if the signer address was not previously approved

Reply:
I had this debate with myself. Here's why I thought it's ok to leave it as-is:

- I expect nearly 100% of the time that senders will only ask to invoke senders they previously approved
- Checking if this sender was approved costs a bit of gas
- To minimize gas costs for the vast majority of use cases I thought to omit this check
- In a small fraction of use cases senders will send signers that were never approved and waste some storage writing gas

So bottom line, I was aiming at the small gas optimization for the main use case.

wdyt?

Reply:
Makes sense. I just wanted to make sure that the state transition from not approved to revocation in progress doesn't present any problems.

I don't think it does. The states for signers are:

- `NOT_APPROVED`
- `APPROVED`
- `REVOCATION_IN_PROGRESS`
- `REVOKED`

In these scenarios, the following state transitions for a signer is possible:

`approveSigners()`: `NOT_APPROVED` -> `APPROVED` OR `REVOKED` -> `APPROVED` OR `REVOCATION_IN_PROGRESS` -> `APPROVED`
`requestSignersRevocation()`: `APPROVED` -> `REVOCATION_IN_PROGRESS` OR `NOT_APPROVED` -> `REVOCATION_IN_PROGRESS`
`block.number >= revocationBlock`: `REVOCATION_IN_PROGRESS` -> `REVOKED`